go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -411,6 +396,28 @@ func (bnc *BaseNetworkController) getExpectedSwitchName(pod *kapi.Pod) (string,
 	return switchName, nil
 }
 
+// ensurePodAnnotation will copy pod annotations at migratable pods related
+// to the same virtual machine, for normal pods it will unmarshal and return
+// it, also there returned boolean will be true if the pod subnet belong to
+// controller's zone.
+func (bnc *BaseNetworkController) ensurePodAnnotation(pod *kapi.Pod, nadName string) (*util.PodAnnotation, bool, error) {
+	if kubevirt.IsPodLiveMigratable(pod) {
+		podAnnotation, err := kubevirt.EnsurePodAnnotationForVM(bnc.watchFactory, bnc.kube, pod, bnc.NetInfo, nadName)
+		if err != nil {
+			return nil, false, err
+		}
+		// The live migrated pods will have the pod annotation but the switch manager running
+		// at the node will not contain the switch as expected
+		_, zoneContainsPodSubnet := kubevirt.ZoneContainsPodSubnet(bnc.lsManager, podAnnotation)
+		return podAnnotation, zoneContainsPodSubnet, nil
+	}
+	podAnnotation, err := util.UnmarshalPodAnnotation(pod.Annotations, nadName)
+	if err != nil {
+		return nil, false, err
+	}
+	return podAnnotation, true, nil
```

Comment:
if `UnmarshalPodAnnotation` return err, should we ignore it, return a nil error and an nil podAnnotation.

And then from the function caller, check return error if this function returns error. And check for nil podAnnotation to the same effect nil was being check for err before.

otherwise, we miight be ignoring a rightful error from `EnsurePodAnnotationForVM`